src/Parallafka/MessageFinishedRouter.cs

```diff
@@ -7,34 +8,64 @@ namespace Parallafka
     internal class MessageFinishedRouter<TKey, TValue>
     {
         private readonly MessagesByKey<TKey, TValue> _messageByKey;
-        private readonly BufferBlock<IKafkaMessage<TKey, TValue>> _messagesToHandle;
+        private readonly BufferBlock<KafkaMessageWrapped<TKey, TValue>> _messagesToHandle;
+
+        private long _messagesHandled;
+        private long _messagesSkipped;
+        private long _messagesSent;
+        private long _messagesNotSent;
 
         public MessageFinishedRouter(MessagesByKey<TKey, TValue> messageByKey)
         {
             this._messageByKey = messageByKey;
-            this._messagesToHandle = new(new DataflowBlockOptions { BoundedCapacity = 1 });
+            this._messagesToHandle = new(new DataflowBlockOptions
+            {
+                BoundedCapacity = 1
+            });
         }
 
-        public ISourceBlock<IKafkaMessage<TKey, TValue>> MessagesToHandle => this._messagesToHandle;
+        public ISourceBlock<KafkaMessageWrapped<TKey, TValue>> MessagesToHandle => this._messagesToHandle;
 
         public Task Completion => this._messageByKey.Completion;
 
+        public object GetStats()
+        {
+            return new
+            {
+                InputCount = this._messagesToHandle.Count,
+                MessagesHandled = this._messagesHandled,
+                MessagesSent = this._messagesSent,
+                MessagesNotSent = this._messagesNotSent,
+                MessagesSkipped = this._messagesSkipped
+            };
+        }
+
         public void Complete()
         {
             this._messageByKey.Complete();
         }
 
-        public async Task MessageHandlerFinished(IKafkaMessage<TKey, TValue> message)
+        public async Task MessageHandlerFinished(KafkaMessageWrapped<TKey, TValue> message)
         {
+            Interlocked.Increment(ref this._messagesHandled);
             // If there are any messages with the same key queued, make the next one available for handling.
             // TODO: Is this safe as far as commits?
             if (this._messageByKey.TryGetNextMessageToHandle(message, out var newMessage))
             {
-                if (!await this._messagesToHandle.SendAsync(newMessage))
+                if (await this._messagesToHandle.SendAsync(newMessage))
+                {
+                    Interlocked.Increment(ref this._messagesSent);
+                }
+                else
                 {
+                    Interlocked.Increment(ref this._messagesNotSent);
                     Parallafka<TKey, TValue>.WriteLine($"MFR: {newMessage.Key} {newMessage.Offset} SendAsync failed!");
                 }
             }
+            else
+            {
+                Interlocked.Increment(ref this._messagesSkipped);
```

Comment:
Not sure about the naming "skipped" here. Doesn't this get called pretty much all the time and it just means there were no more messages with the same key?

Reply:
What should it be called?  Maybe it doesn't even need to be there.